src/main/java/Frontend/CommandLineInterface.java

```diff
@@ -17,12 +17,51 @@ public static void main(String[] args) {
             String firstElementOfPair = userInputs.get(i).toLowerCase();
             String secondElementOfPair = userInputs.get(i + 1);
             if (!Arrays.asList(acceptedCommands).contains(firstElementOfPair)) {
-                System.out.println(firstElementOfPair + " is not a valid command. Please try again.");
+                System.out.println("\"" + firstElementOfPair + "\" is not a valid command. Please try again.");
+                userInputIsValid = false;
                 break;
             } else if (secondElementOfPair == null) {
-                System.out.println(firstElementOfPair + " has no valid response. Please try again.");
+                System.out.println("\"" + firstElementOfPair + "\" has no valid response. Please try again.");
+                userInputIsValid = false;
                 break;
+            } else if (firstElementOfPair.equals("-dim") && !isPositiveInteger(secondElementOfPair)) {
+                System.out.println("-dim needs to be followed by a positive integer. Please try again.");
+                userInputIsValid = false;
+                break;
+            }
+        }
+        // At the point, the user input is at least a (structurally) valid input, with the correct pair of
+        // (command, response).
+        if (userInputIsValid) {
+            System.out.println("Hooray!");
+        }
+    }
+
+    /**
+     * Adapted from https://stackoverflow.com/questions/237159/
+     * whats-the-best-way-to-check-if-a-string-represents-an-integer-in-java. Checks whether an input string
+     * is a valid positive integer.
+     * @param userInput The user input
+     * @return whether the user input is a valid positive integer
+     */
+    private static boolean isPositiveInteger(String userInput) {
```

Comment:
hmm i think instead of having a bunch of static methods and attributes, what you could do is create an instance of CLI in the main method and then call things. Similar to what we do for Constants